### PR TITLE
[History] load history

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot-Commons/pull/320

### CODE REVIEW

File: octobot_trading/personal_data/portfolios/portfolio_manager.py [hunk 1]
```diff
@@ -227,15 +254,30 @@ def _refresh_simulated_trader_portfolio_from_order(self, order):
...
+                if reset_from_config \
+                        or self.historical_portfolio_value_manager is None \
+                        or not self.historical_portfolio_value_manager.has_previous_session_portfolio():
+                    self._set_starting_simulated_portfolio()
+                else:
+                    self._load_simulated_portfolio_from_history()
             self.logger.info(f"{constants.CURRENT_PORTFOLIO_STRING} {self.portfolio.portfolio}")
 
+    def _load_simulated_portfolio_from_history(self):
+        #  todo also load available amounts when loading simulated orders
```
Comment (Herklos) [inline octobot_trading/personal_data/portfolios/portfolio_manager.py:272]: Is it done now with `_apply_locked_assets`?

File: octobot_trading/personal_data/trades/channel/trades_updater.py [hunk 2]
```diff
@@ -69,6 +69,12 @@ async def fetch_and_push(self):
             trades: list = await self.channel.exchange_manager.exchange.get_my_recent_trades(
                 symbol=symbol,
                 limit=self.MAX_OLD_TRADES_TO_FETCH)
+            if not trades:
+                # on some exchanges, recent trades are only fetching very recent trade. also try closed orders
```
Comment (Herklos) [inline octobot_trading/personal_data/trades/channel/trades_updater.py:73]: Why no managing it in the ccxt exchange (or exchange tentacle)? I don't understand why we should fix exchanges behaviors here

File: octobot_trading/personal_data/trades/trades_manager.py [hunk 3]
```diff
@@ -92,6 +96,17 @@ def _reset_trades(self):
         self.trades_initialized = False
         self.trades = collections.OrderedDict()
 
+    async def _load_trades_history(self):
+        if self.trader.exchange_manager.is_backtesting:
+            # don't load history on backtesting
+            return
+        try:
+            if self.trader.exchange_manager.storage_manager.trades_storage:
```
Comment (Herklos) [inline octobot_trading/personal_data/trades/trades_manager.py:104]: Can `exchange_manager.storage_manager` be None ?

File: octobot_trading/personal_data/portfolios/portfolio_manager.py (hunk 1 above)
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/portfolios/portfolio_manager.py:272]: we will be able to do it when simulated orders will be saved in db and reloaded at startup

File: octobot_trading/personal_data/trades/channel/trades_updater.py (hunk 2 above)
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/trades/channel/trades_updater.py:73]: right, i'll change it

File: octobot_trading/personal_data/trades/trades_manager.py (hunk 3 above)
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/trades/trades_manager.py:104]: Nope, it's initialized in exchange manager constructor

Comment (Herklos) [reply octobot_trading/personal_data/trades/trades_manager.py:104]: Even if storage is disabled?

File: octobot_trading/exchanges/connectors/ccxt/ccxt_connector.py [hunk 4]
```diff
@@ -333,7 +333,16 @@ async def get_my_recent_trades(self, symbol: str = None, since: int = None,
         if self.client.has['fetchMyTrades'] or self.client.has['fetchTrades']:
             with self.error_describer():
                 method = self.client.fetch_my_trades if self.client.has['fetchMyTrades'] else self.client.fetch_trades
-                return self.adapter.adapt_trades(await method(symbol=symbol, since=since, limit=limit, params=kwargs))
+                trades = self.adapter.adapt_trades(await method(symbol=symbol, since=since, limit=limit, params=kwargs))
+                if trades:
+                    return trades
+                # on some exchanges, recent trades are only fetching very recent trade. also try closed orders
```
Comment (Herklos) [inline octobot_trading/exchanges/connectors/ccxt/ccxt_connector.py:339]: :+1:

File: octobot_trading/personal_data/trades/trades_manager.py (hunk 3 above)
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/trades/trades_manager.py:104]: when storage is disabled `self.trader.exchange_manager.storage_manager.trades_storage` is None and  `self.trader.exchange_manager.storage_manager` exists. But maybe you mean something else ?
image

Comment (Herklos) [reply octobot_trading/personal_data/trades/trades_manager.py:104]: No it's ok thanks :+1:

**Review by Herklos — APPROVED** [review]
:+1:

**Pull request merged**